src/render_tasks/d3d12_deferred_main.hpp

```diff
@@ -53,6 +54,19 @@ namespace wr
 				auto d3d12_cb_handle = static_cast<D3D12ConstantBufferHandle*>(scene_graph.GetActiveCamera()->m_camera_cb);
 				d3d12::BindConstantBuffer(cmd_list, d3d12_cb_handle->m_native, 0, frame_idx);
 
+				auto texture_pool_raw = static_cast<D3D12TexturePool*>(n_render_system.m_texture_pool.get());
+				
+				if (texture_pool_raw)
+				{
+					texture_pool_raw->Stage(cmd_list);
+
+					wr::d3d12::DescriptorHeap* texture_heap = texture_pool_raw->GetHeap();
```

Comment:
How do you know the user needs this texture pool? The scene graph should determine this.

Reply:
Removed from rendering code, will implement in next branch